sys/net/network_layer/ng_ipv6/ng_ipv6.c

```diff
@@ -594,5 +600,21 @@ static void _receive(ng_pktsnip_t *pkt)
     ng_ipv6_demux(iface, pkt, hdr->nh);
 }
 
+static void _decapsulate(ng_pktsnip_t *pkt)
+{
+    ng_pktsnip_t *ptr = pkt;
+
+    pkt->type = NG_NETTYPE_UNDEF;   /* prevent payload (the encapsulated packet)
+                                     * from being removed */
+
+    /* Remove encapsulating IPv6 header */
+    while ((ptr->next != NULL) && (ptr->next->type == NG_NETTYPE_IPV6)) {
```

Comment:
does this also remove the next headers, which are part of the encapsulating ipv6 header?